### PR TITLE
Fix https://github.com/Azure/azure-cli/issues/3343

### PR DESCRIPTION
**Pull request by vivsriaus**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def get_resources_in_resource_group(resource_group_name, resource_type=None):
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
     rcf = get_mgmt_service_client(ResourceType.MGMT_RESOURCE_RESOURCES)
     filter_str = "resourceType eq '{}'".format(resource_type) if resource_type else None
-    return list(rcf.resource_groups.list_resources(resource_group_name, filter=filter_str))
+    return list(rcf.resources.list_by_resource_group(resource_group_name, filter=filter_str))
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py]: @derekbekoe, is resources one of the modules that must be stack compliant?

Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py]: Yes.
Did the rename happen for the older API versions of Resources as well? If so then no extra logic is needed here.

Comment (vivsriaus) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py]: @derekbekoe the rename wasn't done for older API versions. How do you propose we handle this?

Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py]: Okay then you have to do two things depending on which API version is active.
```
if supported_api_version(ResourceType.MGMT_RESOURCE_RESOURCES, max_api='older_api_version'):
   do the old way.
else:
   do the newer way.
```

**Comment by vivsriaus** [comment]
@tjprescott @derekbekoe Looks good?

**Pull request merged**